src/main/java/no/unit/nva/model/validator/IssnValidator.java

```diff
@@ -0,0 +1,64 @@
+package no.unit.nva.model.validator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class IssnValidator {
+
+    public static final int ISSN_STANDARD_LENGTH = 9;
+    public static final int ISSN_HYPHEN_POSITION = 4;
+    public static final char HYPHEN = '-';
+    public static final int MATCH_TEN = 10;
+
+    /**
+     * Validator for ISSNs, checks the structure and modulo 11 checkbit.
+     * The validator does not check for empty or null values.
+     *
+     * @param issn An ISSN for validation
+     * @return Boolean for the result of the validation
+     */
+    public static boolean validate(String issn) {
+        if (issn.length() != ISSN_STANDARD_LENGTH) {
+            return false;
+        }
+        return testStructure(issn);
+    }
+
+    private static boolean testStructure(String issn) {
+        String value = issn.toUpperCase(Locale.ENGLISH);
+        char[] characters = value.toCharArray();
+        List<Integer> integers = new ArrayList<>();
+        for (int counter = 0; counter < characters.length - 1; counter++) {
+            if (counter == ISSN_HYPHEN_POSITION && characters[counter] != HYPHEN) {
+                return false;
+            }
+            int number;
+            if (counter != ISSN_HYPHEN_POSITION) {
+                try {
+                    number = Integer.parseInt(String.valueOf(characters[counter]));
+                } catch (NumberFormatException e) {
+                    return false;
+                }
+                integers.add(number);
+            }
+        }
+        return generateCheckbit(integers) == characters[characters.length - 1];
```

Comment:
refactor to a method with a meaningful name

Reply:
I wonder what the name for generation of a checkbit would be? :D

Reply:
Maybe he means the statement itself.